src/query/plan/rule_based_planner.hpp

```diff
@@ -567,6 +565,66 @@ class RuleBasedPlanner {
                                            symbol);
   }
+
+  std::unique_ptr<LogicalOperator> GenFilters(std::unique_ptr<LogicalOperator> last_op,
+                                              const std::unordered_set<Symbol> &bound_symbols, Filters &filters,
+                                              AstStorage &storage, const SymbolTable &symbol_table) {
+    std::unique_ptr<LogicalOperator> pattern_filter =
+        ExtractPatternFilters(filters, storage, symbol_table, bound_symbols);
+    auto *filter_expr = impl::ExtractFilters(bound_symbols, filters, storage);
+    if (filter_expr) {
+      last_op = std::make_unique<Filter>(std::move(last_op), std::move(pattern_filter), filter_expr);
+    }
+    return last_op;
+  }
+
+  std::unique_ptr<LogicalOperator> ExtractPatternFilters(const Filters &filters, AstStorage &storage,
+                                                         const SymbolTable &symbol_table,
+                                                         const std::unordered_set<Symbol> &bound_symbols) {
+    for (const auto &filter : filters) {
+      if (filter.type != FilterInfo::Type::Pattern) {
+        continue;
+      }
+
+      if (const auto *exists = utils::Downcast<Exists>(filter.expression)) {
+        return MakeExistsFilter(*exists, storage, symbol_table, bound_symbols);
+      }
+
+      throw SemanticException("Pattern filter does not exist!");
+    }
+
+    return nullptr;
+  }
+
+  std::unique_ptr<LogicalOperator> MakeExistsFilter(const Exists &exists, AstStorage &storage,
+                                                    const SymbolTable &symbol_table,
+                                                    const std::unordered_set<Symbol> &bound_symbols) {
+    auto *from_node = static_cast<NodeAtom *>(exists.pattern_->atoms_[0]);
+    auto *relationship = static_cast<EdgeAtom *>(exists.pattern_->atoms_[1]);
+    auto *to_node = static_cast<NodeAtom *>(exists.pattern_->atoms_[2]);
+    auto node1_symbol = symbol_table.at(*from_node->identifier_);
+    auto node2_symbol = symbol_table.at(*to_node->identifier_);
+    auto edge_symbol = symbol_table.at(*relationship->identifier_);
+    auto direction = relationship->direction_;
+
+    std::vector<storage::EdgeTypeId> edge_types;
+    edge_types.reserve(relationship->edge_types_.size());
+    for (const auto &type : relationship->edge_types_) {
+      edge_types.push_back(GetEdgeType(type));
+    }
+
+    std::vector<Symbol> once_symbols(bound_symbols.begin(), bound_symbols.end());
+    std::unique_ptr<LogicalOperator> last_op = std::make_unique<Once>(once_symbols);
+
+    last_op = std::make_unique<Expand>(std::move(last_op), node1_symbol, node2_symbol, edge_symbol, direction,
+                                       edge_types, false, storage::View::OLD);
+
+    last_op = std::make_unique<Limit>(std::move(last_op), storage.Create<IntegerLiteral>());
```

Comment:
I am somewhat confident we can use here Identifier which will read from frame and return 1.

Reply:
Here we have a definition of an Identifier and what it does.

https://memgraph.notion.site/Semantic-Analysis-and-Symbol-Generation-57b5942e8165476da91f499be71fe326

If we create an anonymous symbol on the frame and store in it TypedValue(1), we can get such a symbol here, maybe, and use Identifier as Expression.